AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/editor_component/editor_script_canvas.py

```diff
@@ -93,18 +104,19 @@ def set_variable_value(self, variable_name: str, variable_state: VariableState,
         param variable_name: the name of the variable
         param variable_state: used/unused variable
         param variable_value: the value to assign to the variable
+        param component_index: the index of the script canvas component (entities can have multiple script canvas components)
 
         returns None
         """
         component_property_path = self.__construct_variable_component_property_path(variable_name, variable_state)
-        print(variable_value)
-        self.script_canvas_component.set_component_property_value(component_property_path, variable_value)
 
-        #validate the change
-        set_value = self.script_canvas_component.get_component_property_value(component_property_path)
-        assert set_value == variable_value, f"Component variable {variable_name} was not set properly"
+        from editor_python_test_tools.editor_component.editor_component_validation import (
+            validate_script_canvas_variable_changed)
 
-    def __construct_variable_component_property_path(self, variable_name: str, variable_state: VariableState) -> str:
+        validate_script_canvas_variable_changed(self.get_script_canvas_components, component_property_path,
+                                                variable_name, variable_value, component_index)
```

Comment:
As discussed remove the validation from the setters and move them to the test steps.